subgoals/sg_generating.py

```diff
@@ -0,0 +1,514 @@
+                if act.strip() == steps[num_act]['action']:
+                    st = steps[num_act]
+                    st['subgoal'] = subgoal
+                    print(" current detected sg is :", subgoal)
+                    valid_actions.append(act)
+                    if not added_sg:
+                        subgoals_list_insteps.append(subgoal)
+                        print(" subgoals_list_insteps : ", subgoals_list_insteps)
+                        added_sg = True
+                        modified_answer +=  str(len(subgoals_list_insteps))+ '- ' + subgoal + ': {'
+                    modified_answer += act + ', '
+                    new_steps.append(st)
+                    num_act += 1
+                    if num_act == len(steps):
+                        break
+                else:
+                    continue
+            if added_sg:
+                modified_answer += '}\n'
+                item = {'action':'end subgoal', 'observation':'the current subgoal is done.', 'score':steps[num_act-1]['score'], 'isCompleted':steps[num_act-1]['isCompleted'], 'freelook':steps[num_act-1]['freelook'], 'inventory': steps[num_act-1]['inventory'], 'subgoal':subgoal}
+                new_steps.append(item)
+
+
+
+    return new_steps, subgoals_list_insteps, modified_answer
+                
+def sg_extraction_completion(instr):
+    subgoals_list = []
+    actions_list = [] ## list of all actions
+    subgoals = []
+    num_action = 0
+    subgoal_action_dict = {}
+    subgoal_action_index_list = []
+
+    for i in range(1, 40):
+        if str(i)+'-' in instr:
+            st = instr.find(str(i)+'-')+3
+            if str(i + 1) + '-' in instr:
+                end = instr.find(str(i + 1) + '-')
+                sg = instr[st:end]
+            else:
+                sg = instr[st:]
+            if ':{' in sg:
+                action = sg[sg.find(':{') + 2:-2].split(',')
+                subgoal = sg[: sg.find(':{')]
+            elif ': {' in sg:
+                action = sg[sg.find(': {') + 3:-2].split(',')
+                subgoal = sg[: sg.find(': {')]
+            action = [re.sub('[{,}.]', '', act) for act in action]
+            if '' in action:
+                action.remove('')
+
+            subgoals_list.extend(subgoal for i in range(len(action)))
+
+            ### add completed subgoals
+            subgoals.append(subgoal)
+            subgoal_action_dict[subgoal] = action
+            ### align subgoals to actions in steps
+            for act in action:
+                act = act.replace("'", "").strip()
+                actions_list.append(act)
+                num_action += 1
+                subgoal_action_index_list.append({'action':act, 'sg':subgoal, 'idx': num_action-1})
+
+    return actions_list, subgoal_action_index_list
+
+def check_action_sg(actions_list, gold_path, sg_act_idx_list, task_desc, chatgpt_answer):
+    print(f'len goal path {len(gold_path)} ---- len action seq {len(actions_list)}')
```

Comment:
again, adding a short docstring could be helpful.

Reply:
I have added them.